src/utilities/modules/vaccines/aggregation.js

```diff
@@ -0,0 +1,212 @@
+import {
+  linkSensorLogToItemBatches,
+  createFullAggregateSensorLogs,
+  createGenericSensorLog,
+} from './utilities';
+
+/**
+ * Aggregation methods for SensorLogs. PreAggregate
+ * and FullAggregate.
+ *
+ * Preaggregates are aggregations of raw data synced
+ * from a sensor. Sensors can be quite sensitive to
+ * temperatures, so a shorter interval is used on a
+ * physical sensor to achieve a more representitive
+ * temperature once aggregated.
+ *
+ * Full aggregations are much larger consolidations
+ * of sensor logs in which a breach has not occurred,
+ * to limit the number of SensorLog records as well
+ * as SensorLogItemBatchJoins.
+ */
+
+const PREAGGREGATE_TYPE = 'preAggregate';
+const FULL_AGGREGATE_TYPE = 'aggregate';
+
+const ONE_MINUTE_MILLISECONDS = 1000 * 60;
+const ONE_HOUR_MILLISECONDS = ONE_MINUTE_MILLISECONDS * 60;
+const FIVE_MINUTES_MILLISECONDS = ONE_MINUTE_MILLISECONDS * 5;
+const EIGHT_HOURS_MILLISECONDS = ONE_HOUR_MILLISECONDS * 8;
+
+const NO_FULL_AGGREGATE_PERIOD = EIGHT_HOURS_MILLISECONDS * 4 + FIVE_MINUTES_MILLISECONDS;
+const FULL_AGGREGATION_INTERVAL = EIGHT_HOURS_MILLISECONDS + FIVE_MINUTES_MILLISECONDS;
+const PREAGGREGATE_INTERVAL = 20 * ONE_MINUTE_MILLISECONDS;
+/**
+ * Aggregates sensor logs that have been pulled from a physical sensor.
+ */
+export function preAggregateLogs({ sensor, database }) {
```

Comment:
we'll have to remember that aggregation should be based on location, i..e sensor independent (maybe add a todo here, or i'll try to remember to add an issue)